--- a/Postal_TBC.toc
+++ b/Postal_TBC.toc
@@ -1,4 +1,4 @@
-## Interface: 20504
+## Interface: 20506
 ## Title: Postal
 ## Notes: Postal: Enhanced Mailbox support
 ## Author: Xinhuan (BlackBook), Aery (QuickAttach), Marilin (Express), Gnarfoz (Rake), grennon (Select), arith (CarbonCopy), Nevcairiel (OpenAll, Wire), Tuller (TradeBlock), Rabbit (DoNotWant), Phanx (Express), lhunath (Express), Antiarc (Express), Stanzilla (Express), and others
